misk/src/main/kotlin/misk/web/WebConfig.kt

```diff
@@ -114,10 +114,15 @@ data class WebConfig @JvmOverloads constructor(
   val close_connection_percent: Double = 0.0,
 
   /**
-   * If true responses which are larger than the minGzipSize will be compressed.
+   * If true non-gRPC responses which are larger than the minGzipSize will be compressed.
    */
   val gzip: Boolean = true,
 
+  /**
+   * If true gRPC responses which are larger than the minGzipSize will be compressed.
+   */
+  val grpcGzip: Boolean = false,
```

Comment:
Should this instead be an enum to allow for the other wire-supported encoding? 

```suggestion
  val grpcEncoding: GrpcEncoding = GrpcEncoding.IDENTITY,
```

Reply:
Yes, I'd lean towards the encoding enum, not just a boolean

Reply:
Not sure if it's actually worth it:
1. The client only supports `identity` and `gzip`: https://github.com/square/wire/blob/master/wire-grpc-client/src/commonMain/kotlin/com/squareup/wire/internal/GrpcDecoder.kt
2. The http response compression only supports gzip: https://github.com/cashapp/misk/blob/4da16a5c6492d5fa5b0364d95686cdaef6d692e5/misk/src/main/kotlin/misk/web/WebConfig.kt#L119
3. We also need to introduce `WebConfig.minGrpcMessageToCompress` instead of reusing `WebConfig.minGzipSize`.

Reply:
I think the current implementation makes more sense to the user:

1. Why do you configure `grpcEncoding` to anything that's not supported?
2. Why do you want different configs for min compression size for http vs grpc over http?

cc @Hexcles

Reply:
Personally, I'd say @zpingcai 's current impl is better in terms of consistency.

`grpcGzip` makes a lot of sense along with `gzip`. If we want `grpcEncoding`, I'd expect to have an enum for plain old HTTP, too. And in fact, not just an enum but a config object as different compression algos have different knobs (`minGzipSize` obviously doesn't make sense for non-gzip compression algos).